src/prng.rs

```diff
@@ -20,6 +20,140 @@ pub enum PrngError {
     Suite(#[from] SuiteError),
 }
 
+/// This type implements an iterator that generates a pseudorandom sequence of field elements. The
+/// sequence is derived from the key stream of AES-128 in CTR mode with a random IV.
+#[derive(Debug)]
+pub(crate) struct Prng3<F> {
+    phantom: PhantomData<F>,
+    key_stream: KeyStream,
+    buffer: Vec<u8>,
+    buffer_index: usize,
+    output_written: usize,
+}
+
+impl<F: FieldElement> Prng3<F> {
+    /// Generates a seed and constructs an iterator over an infinite sequence of pseudorandom field
+    /// elements.
+    pub(crate) fn generate(suite: Suite) -> Result<Self, PrngError> {
+        let key = Key::generate(suite)?;
+        let key_stream = KeyStream::from_key(&key);
+        Ok(Self::from_key_stream(key_stream))
+    }
+
+    pub(crate) fn from_key_stream(mut key_stream: KeyStream) -> Self {
+        let mut buffer = vec![0; BUFFER_SIZE_IN_ELEMENTS * F::ENCODED_SIZE];
+        key_stream.fill(&mut buffer);
+
+        Self {
+            phantom: PhantomData::<F>,
+            key_stream,
+            buffer,
+            buffer_index: 0,
+            output_written: 0,
+        }
+    }
+
+    pub(crate) fn get(&mut self) -> F {
+        loop {
+            // Seek to the next chunk of the buffer that encodes an element of F.
+            for i in (self.buffer_index..self.buffer.len()).step_by(F::ENCODED_SIZE) {
+                let j = i + F::ENCODED_SIZE;
+                if let Some(x) = match F::hash_to_field(&self.buffer[i..j]) {
```

Comment:
It looks like what we're doing here is generating a pseudorandom string from AES-CTR, then applying hash_to_field directly to the output. This isn't what I intended for us to compare. The fact that this isn't initially clear is my fault. I'm sorry about that. Let me try to be more clear.

What we want to measure is the performance of the algorithm specified [here](https://github.com/cjpatton/vdaf/pull/29), which transforms a byte string into a sequence of field elements. This algorithm does not apply hash_to_field directly, but rather a variant that replaces `expand_message(msg, DST, len_in_bytes)` (see https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-13.html#name-hash_to_field-implementatio) with the algorithm currently used in libprio for expanding a seed into a byte string of the requested length.

In other words, `random_vector3` should look something like this:
```rust
pub fn random_vector3<F: FieldElement>(len: usize, padding: usize) -> Result<Vec<F>, PrngError> {
    let key = Key::generate(Suite::Aes128CtrHmacSha256)?;
    let key_stream = KeyStream::from_key(&key);
    let mut buf = vec![0; len * (F::ENCODED_SIZE + padding)];
    key_stream.fill(&mut buf);
    Ok(buf.chunks(F::ENCODED_SIZE + padding).map(|chunk| F::from_be_bytes(chunk).unwrap()).collect())
}
```

(By the way, there's no need to re-implement the `Prng` API. We'll probably want something like this later, but I suggest we keep it simple for now.)

Reply:
your snippet is much simpler, now is refactored.